### PR TITLE
feat(iot-device): Add support for PnP by passing in the DCM ID

### PR DESCRIPTION
**Pull request by abhipsaMisra**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

File: iothub/device/src/ClientOptions.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// Options that allow configuration of the device client instance during initialization.
+    /// </summary>
+    public class ClientOptions
+    {
+        /// <summary>
+        /// The Digital Twin Model Id associated with the device client instance.
```
Comment (drwill-ms) [inline iothub/device/src/ClientOptions.cs:12]: Digital Twins?

Comment (abhipsaMisra) [reply iothub/device/src/ClientOptions.cs:12]: sure, should it also be lower case?

Comment (drwill-ms) [reply iothub/device/src/ClientOptions.cs:12]: yes

**Pull request merged**